lib/model/store.dart

```diff
@@ -13,6 +13,50 @@ import '../api/route/messages.dart';
 import '../credential_fixture.dart' as credentials;
 import 'message_list.dart';
 
+/// Store for the user's cross-account data.
+///
+/// This includes data that is independent of the account, like some settings.
+/// It also includes a small amount of data for each account: enough to
+/// authenticate as the active account, if there is one.
+class GlobalStore extends ChangeNotifier {
+  GlobalStore._({required Map<int, Account> accounts})
+      : _accounts = accounts;
+
+  // For convenience, a number we won't use as an ID in the database table.
+  static const fixtureAccountId = -1;
+
+  // We keep the API simple and synchronous for the bulk of the app's code
+  // by doing this loading up front before constructing a [GlobalStore].
+  static Future<GlobalStore> load() async {
+    const accounts = {fixtureAccountId: _fixtureAccount};
+    return GlobalStore._(accounts: accounts);
+  }
+
+  final Map<int, Account> _accounts;
+
+  // TODO settings (those that are per-device rather than per-account)
+  // TODO push token, and other data corresponding to GlobalSessionState
+
+  // Just an Iterable, not the actual Map, to avoid clients mutating the map.
+  // Mutations should go through the setters/mutators below.
+  Iterable<Account> get accounts => _accounts.values;
+
+  Account? getAccount(int id) => _accounts[id];
+
+  // TODO add setters/mutators; will want to write to database
+  // Future<void> insertAccount...
+  // Future<void> updateAccount...
+
+  // TODO add a registry of [PerAccountStore]s, like the latter's of [MessageListView]
+  //   That will allow us to have many [PerAccountRoot] widgets for a given
+  //   account, e.g. at the top of each page; and to access server data from
+  //   outside any [PerAccountRoot], e.g. for handling a notification.
```

Comment:
> many [PerAccountRoot] widgets for a given account, e.g. at the top of each page

This seems like a case where `PerAccountRoot`'s instruction to build a "loading page" could be problematic. Until the store is available, each page will have a loading page at the top?

I don't yet understand why we'll want to have multiple `PerAccountRoot`s per account. In what sense is each one a "root", then? 🤔

Reply:
> Until the store is available, each page will have a loading page at the top?

Hmm, I suppose there's more for me to think through on that front. (But this is a TODO describing #21, not really part of what this PR itself does — so I think it needn't block merging this one.)


> In what sense is each one a "root", then? :thinking: 

I think as part of this the name will change to not say "root".


> I don't yet understand why we'll want to have multiple `PerAccountRoot`s per account.

Does #21 explain it? If not, let's discuss there.